### DISCUSSION TITLE
v1.0.0

### DISCUSSION THREAD

**Discussion by DomCR** [DomCR/ACadSharp#435] [Announcements]
## What's Changed
* Bounding box by @DomCR in https://github.com/DomCR/ACadSharp/pull/213
* Code coverage by @DomCR in https://github.com/DomCR/ACadSharp/pull/404
* Issue 407 fix true color value by @DomCR in https://github.com/DomCR/ACadSharp/pull/410
* Dwg entity mesh by @DomCR in https://github.com/DomCR/ACadSharp/pull/341
* DwgWriter fix for Linetypes with segments by @DomCR in https://github.com/DomCR/ACadSharp/pull/414
* Issue 409 text encoding fix by @DomCR in https://github.com/DomCR/ACadSharp/pull/415
* Issue 369 pdfdefinition by @DomCR in https://github.com/DomCR/ACadSharp/pull/370
* Coverage test by @DomCR in https://github.com/DomCR/ACadSharp/pull/420
* repo-rescturcture by @DomCR in https://github.com/DomCR/ACadSharp/pull/416
* wiki generation fix by @DomCR in https://github.com/DomCR/ACadSharp/pull/423
* header color index fix by @DomCR in https://github.com/DomCR/ACadSharp/pull/424
* Issue-421_dxf-irregular-no-header by @DomCR in https://github.com/DomCR/ACadSharp/pull/425
* Issue 417 hatch polyline by @DomCR in https://github.com/DomCR/ACadSharp/pull/427
* Cadcollections refactor by @DomCR in https://github.com/DomCR/ACadSharp/pull/429
* Sort entities table fix by @DomCR in https://github.com/DomCR/ACadSharp/pull/430
* Issue-405 Implement Attribute Multiline by @DomCR in https://github.com/DomCR/ACadSharp/pull/431
* Unknown None Graphical Object by @DomCR in https://github.com/DomCR/ACadSharp/pull/428
* Road to Version 1.0.0 by @DomCR in https://github.com/DomCR/ACadSharp/pull/288

**Full Changelog**: https://github.com/DomCR/ACadSharp/compare/v2.4.0-beta...v1.0.0

[NuGet Package](https://www.nuget.org/packages/ACadSharp)

<hr /><em>This discussion was created from the release <a href='https://github.com/DomCR/ACadSharp/releases/tag/v1.0.0'>v1.0.0</a>.</em>

**Comment by DomCR** [comment]
# v1.0.0 Release:

V1 is here! The project has been stable enough and all the main features have been implemented but this doesn't end here:

## Roadmap:

### Missing entities/objects:
There are still some missing entities and objects to implement, for all of them if the reader cannot identify the entity or object it can store some of the information as an `Unknown` these entities will be implemented in future patches, like:

Entities:
- https://github.com/DomCR/ACadSharp/pull/314 
- https://github.com/DomCR/ACadSharp/pull/191
- Proxies (not all of them are documented or identified)

Objects:
- Section
- TableStyle
- VisualStyle
- Material
- Dimassoc
- Geodata
- Proxy Object

### Features

Some features are planned to be added after the release:
- https://github.com/DomCR/ACadSharp/pull/283
- https://github.com/DomCR/ACadSharp/pull/211
- Geometric transformation (math utils)
- Export to Json (will allow to export a `CadDocument` into a Json format to be used instead of saving the information to a dxf or dwg)
- ... (suggestions?)

### Related Projects

Another important feature to be implemented is to create/export the dxf/dwg drawings to a pdf, I've started this new project to implement this feature:

- **https://github.com/DomCR/ACadSharp.Pdf**

The feature is separated form the main project to avoid loading unnecessary dependencies, `ACadSharp.Pdf` uses the open-source `PdfSharp` to create the pdf output and other libraries may need to be added.

**Reply by Hexer611** [reply]
Congratz for V1 of the repo. Great work!

**Comment by riderking** [comment]
Great work!
